packages/flutter/lib/src/material/tab_controller.dart

```diff
@@ -373,6 +373,10 @@ class DefaultTabController extends StatefulWidget {
   /// match [TabBar.tabs]'s and [TabBarView.children]'s length.
   ///
   /// The [initialIndex] argument must not be null.
+  ///
+  /// See also:
+  ///
+  ///  * [PageStorageKey] are used to remember the index.
   const DefaultTabController({
```

Comment:
DefaultTabController seems like an odd place to implement this feature. What if I use a TabController directly? Don't I want this there as well? Similarly to how it is also implemented in PageController and ScrollController....

Reply:
In the #10969 example, by simply placing the PageController outside of the pageview, we don't need to do any additional behavior to ensure that the behavior is consistent with the current PR target behavior.

Reply:
Yes,  but the fix should also work in the other situations.